File: lesson_13/maps_java/maps_app/src/main/java/com/codedifferently/lesson13/Lesson13.java
package com.codedifferently.lesson13;
import java.util.HashMap;

public class Lesson13 {

  /**
   * Provide the solution to LeetCode 3146 here:
   * https://leetcode.com/problems/permutation-difference-between-two-strings
   */
  public int findPermutationDifference(String s, String t) {

 HashMap<Character, Integer> indexFinder = new HashMap<>();

        
        for (int i = 0; i < s.length(); i++) {
            char character = s.charAt(i);
            indexFinder.put(character, i);
        }

        int sum = 0;  

        for (int i = 0; i < t.length(); i++) {
            char character = t.charAt(i); 
            int value = indexFinder.get(character); 
            sum += Math.abs(value - i); 
        }

        return sum; 
    }
  }

File: lesson_13/maps_ts/src/lesson13.ts
/**
 * Provide the solution to LeetCode 3146 here:
 * https://leetcode.com/problems/permutation-difference-between-two-strings
 */
export function findPermutationDifference(s: string, t: string): number {
  const map = new Map<string, number>();

  for (let i = 0; i < t.length; i++) {
    map.set(t[i], i);
  }

  let difference = 0;
  for (let i = 0; i < s.length; i++) {
    const idx = map.get(s[i]);
    if (idx != undefined) {
      difference += Math.abs(i - idx);
    }
  }

  return difference;
}
